src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -108,45 +106,46 @@ template Batch
 
 -- | HIDE
 -- Partially order instructions, so that pass-through chains can be executed in order.
-orderPassThroughChains : [(RoutedStep, InstructionKey)] -> Update [Instruction]
-orderPassThroughChains routedStepsWithInstructionKeys =
+orderPassThroughChains : [(RoutedStep, InstructionKey)] -> Update [InstructionKey]
+orderPassThroughChains routedStepsWithInstructions =
   reverse . fst <$> foldlA
-    (\(orderedInstructions, usedInstructionIds) (routedStep, instructionKey) ->
-      if S.member instructionKey.id usedInstructionIds
+    (\(orderedInstructions, usedInstructions) (routedStep, instructionKey) ->
+      if S.member instructionKey usedInstructions
       then
         -- instruction has already been visited, do not re-insert in list
-        pure (orderedInstructions, usedInstructionIds)
+        pure (orderedInstructions, usedInstructions)
       else do
         (instructionCid, instruction) <- fetchByKey @Instruction instructionKey
         assertMsg "routed step must match" $ routedStep == instruction.routedStep
         let
-          orderedInstructions' = instruction :: orderedInstructions
-          usedInstructionIds' = S.insert instruction.id usedInstructionIds
+          orderedInstructions' = instructionKey :: orderedInstructions
```

Comment:
Above we are switching from `routedStepWithInstructionKeys` to `routedStepWithInstruction`.
Here we switch from `instruction` to `instructionKey`. 

I would try to use consistent naming to indicate the `InstructionKey` (either `instruction` or `instructionKey`)

Reply:
I changed it